src/VisualStudio/Core/Def/CallHierarchy/CallHierarchyCommandHandler.cs

```diff
@@ -39,61 +46,75 @@ internal class CallHierarchyCommandHandler : ICommandHandler<ViewCallHierarchyCo
         [SuppressMessage("RoslynDiagnosticsReliability", "RS0033:Importing constructor should be [Obsolete]", Justification = "Used in test code: https://github.com/dotnet/roslyn/issues/42814")]
         public CallHierarchyCommandHandler(
             IThreadingContext threadingContext,
+            IUIThreadOperationExecutor threadOperationExecutor,
+            IAsynchronousOperationListenerProvider asynchronousOperationListenerProvider,
             [ImportMany] IEnumerable<ICallHierarchyPresenter> presenters,
             CallHierarchyProvider provider)
         {
             _threadingContext = threadingContext;
+            _threadOperationExecutor = threadOperationExecutor;
+            _listener = asynchronousOperationListenerProvider.GetListener(FeatureAttribute.CallHierarchy);
             _presenter = presenters.FirstOrDefault();
             _provider = provider;
         }
 
         public bool ExecuteCommand(ViewCallHierarchyCommandArgs args, CommandExecutionContext context)
         {
-            using (var waitScope = context.OperationContext.AddScope(allowCancellation: true, EditorFeaturesResources.Computing_Call_Hierarchy_Information))
+            var document = args.SubjectBuffer.CurrentSnapshot.GetFullyLoadedOpenDocumentInCurrentContextWithChanges(
+                context.OperationContext, _threadingContext);
+            if (document == null)
             {
-                var cancellationToken = context.OperationContext.UserCancellationToken;
-                var document = args.SubjectBuffer.CurrentSnapshot.GetFullyLoadedOpenDocumentInCurrentContextWithChanges(
-                    context.OperationContext, _threadingContext);
-                if (document == null)
-                {
-                    return true;
-                }
+                return true;
+            }
+
+            var caretPosition = args.TextView.Caret.Position.BufferPosition.Position;
+
+            // We're showing our own UI, ensure the editor doesn't show anything itself.
+            context.OperationContext.TakeOwnership();
+            var token = _listener.BeginAsyncOperation(nameof(ExecuteCommand));
+            ExecuteCommandAsync(document, caretPosition)
+                .ReportNonFatalErrorAsync()
+                .CompletesAsyncOperation(token);
```

Comment:
🙂As we discussed offline, you need to wait for the disposal of the token somewhere in the test code

Reply:
nit: consider stating explicitly that this is firing-and-forgetting so that we can yield the command immediately and allow vs to be responsive while we're doing our work.